letsencrypt/scripts/main.py

```diff
@@ -26,75 +26,78 @@
-        help=config_help("key_dir"))
-    add("--cert-dir", default="/etc/letsencrypt/certs",
-        help=config_help("cert_dir"))
-
-    add("--le-vhost-ext", default="-le-ssl.conf",
-        help=config_help("le_vhost_ext"))
-    add("--cert-path", default="/etc/letsencrypt/certs/cert-letsencrypt.pem",
-        help=config_help("cert_path"))
-    add("--chain-path", default="/etc/letsencrypt/certs/chain-letsencrypt.pem",
-        help=config_help("chain_path"))
+    # Certificate generation
+    apply_on_all(GENERATE_PARSERS, "add_argument", "-k", "--authkey", type=read_file,
+        help="Path to the authorized key file")
+    apply_on_all(GENERATE_PARSERS, "add_argument", "-B", "--rsa-key-size", type=int, default=2048, metavar="N",
+        help=config_help("rsa_key_size"))
+    apply_on_all(GENERATE_PARSERS, "add_argument", "-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    
+    # Revocation
+    revoke_parser.add_argument("--revoke-key", dest="rev_key", type=read_file,
+        help="Revoke all certs generated by the provided authorized key.")
+    revoke_parser.add_argument("--revoke-certificate", dest="rev_cert", type=read_file,
+        help="Revoke a specific certificate.")
+    revoke_parser.add_argument("--no-confirm", dest="no_confirm", action="store_true",
+        help="Turn off confirmation screens")
 
-    add("--apache-server-root", default="/etc/apache2",
+   
+    # Certificate configuration
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--apache-server-root", default="/etc/apache2",
         help=config_help("apache_server_root"))
-    add("--apache-mod-ssl-conf", default="/etc/letsencrypt/options-ssl.conf",
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--apache-mod-ssl-conf", default="/etc/letsencrypt/options-ssl.conf",
         help=config_help("apache_mod_ssl_conf"))
-    add("--apache-ctl", default="apache2ctl", help=config_help("apache_ctl"))
-    add("--apache-enmod", default="a2enmod", help=config_help("apache_enmod"))
-    add("--apache-init-script", default="/etc/init.d/apache2",
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--apache-ctl", default="apache2ctl", help=config_help("apache_ctl"))
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--apache-enmod", default="a2enmod", help=config_help("apache_enmod"))
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--apache-init-script", default="/etc/init.d/apache2",
         help=config_help("apache_init_script"))
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "--le-vhost-ext", default="-le-ssl.conf",
+        help=config_help("le_vhost_ext"))
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    apply_on_all(CONFIGURATION_PARSERS, "add_argument", "-c", "--view-config-changes", action="store_true",
```

Comment:
I vote for removing the short argument entirely in this case. The previous short argument was -v which many users will confuse for `verbose` or possibly `version`